backend/homework/1/data/src/main/java/org/example/StudentRepository.java

```diff
@@ -0,0 +1,42 @@
+package org.example;
+
+import java.util.ArrayList;
+
+public class StudentRepository {
+    ArrayList<Student> array = new ArrayList<Student>();
+//    Student st=new Student(1,"Arunain",21,"A");
+    public void addStudent(Student st){
+        array.add(st);
+    }
+
+    public Student retrieveStudent(int id){
+        for(Student v:array){
+            int currentid=v.getId();
+            if(currentid==id) return v;
+        }
+        System.out.println("Please enter valid id");
+        return null;
+    }
+
+    public Student retrieveStudent(int id, String name){
+        for(Student v:array){
+            int currentid=v.getId();
+            String currentname=v.getName();
+            if(currentid==id && currentname.equals(name)) return v;
+        }
+        System.out.println("Please enter valid id");
+        return null;
+    }
+
+    public void updateStudent(Student updatedStudent) {
```

Comment:
1. Will just one function with args as Student object sufficient for `updateStudent`? Thinking from areal-world scenario.

2. What if user enters invalid data?
What if the object passed is not found?

Shouldn't validations and null checks be made necessary?